File: GeneratedEntityReaderClasses/Components/VerletWorldJointComponent.cs
namespace GeneratedNoitaClasses;
public class VerletWorldJointComponent : NoitaComponentBase,  iNoitaType<VerletWorldJointComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaCVector2<NoitaFloat> world_position { get; set; } = new();
	public NoitaInt verlet_point_index { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		world_position.Read(s);
		verlet_point_index.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		world_position.Write(s);
		verlet_point_index.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/VerletWorldJointComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaCVector2
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class VerletWorldJointComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    world_position:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    verlet_point_index:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.world_position.Read(s)
        self.verlet_point_index.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.world_position.Write(s)
        self.verlet_point_index.Write(s)

